.github/workflows/gen-binaries.yml

```diff
@@ -0,0 +1,138 @@
+name: Generate embedded binaries
+
+on:
+  push:
+    branches:
+      - develop
+
+jobs:
+  gen-nodetime:
+    name: "Generate nodetime binaries"
+    runs-on: ubuntu-latest
+    concurrency: gen-nodetime
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - uses: MOZGIII/install-ldid-action@v1
+        with:
+          tag: v2.1.5-procursus2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - run: ./scripts/gen-nodetime
+
+      - name: Create Pull Request
+        id: cpr
+        uses: peter-evans/create-pull-request@v4
+        with:
+          title: "feat(nodetime): update binaries"
+          commit-message: "feat(nodetime): update binaries"
+          body: ""
+          branch: feat/gen-nodetime
+
+  gen-protoc:
+    name: "Generate protoc binaries"
+    runs-on: ${{ matrix.runner.runs-on }}
+    concurrency: gen-protoc-${{ matrix.runner.os }}-${{ matrix.runner.arch }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner:
+          - runs-on: ubuntu-latest
+            os: linux
+            defaults-shell: bash
+            arch: amd64
+          - runs-on: [ self-hosted, linux, arm64 ]
+            os: linux
+            defaults-shell: bash
+            arch: arm64
+          - runs-on: [ self-hosted, macOS ]
+            os: darwin
+            defaults-shell: /usr/bin/arch -arch x86_64 /bin/bash -l {0}
+            arch: amd64
+          - runs-on: [ self-hosted, macOS ]
+            os: darwin
+            defaults-shell: /usr/bin/arch -arch arm64e /bin/bash -l {0}
+            arch: arm64
+    defaults:
+      run:
+        shell: ${{ matrix.runner.defaults-shell }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - run: |
+          if [[ "${{ matrix.runner.os }}" == "darwin" ]]; then
+            brew install jq autoconf automake libtool
+          else
+            sudo apt install -y jq autoconf automake libtool curl make g++ unzip
+          fi
+
+      - name: Build protoc files
+        run: ./scripts/gen-protoc
+
+      - name: Create Pull Request
```

Comment:
The way this is set up, it will trigger one PR per binary. It should be somehow possible to save each binary as an artifact or cache and create a separate job that picks up the changes and makes a single PR. I did in fact try that, but the job got very messy and error-prone (especially with cache not being warm it did not always pick up all the changes and so on).

Most of the time you won't get new PRs very often, but the first time this runs it will probably trigger 9 PRs in total (1 for nodetime, 4 for protoc and 4 for protoc-gen-dart)